app/spike/internal/cmd/secret/list.go

```diff
@@ -35,11 +37,15 @@ import (
 //	- secret/path3
 //
 // Note: Requires an initialized SPIKE system and valid authentication
-func newSecretListCommand(source *workloadapi.X509Source) *cobra.Command {
+func newSecretListCommand(
+	source *workloadapi.X509Source, spiffeId string,
+) *cobra.Command {
 	var listCmd = &cobra.Command{
 		Use:   "list",
 		Short: "List all secret paths",
 		Run: func(cmd *cobra.Command, args []string) {
+			trust.Authenticate(spiffeId)
```

Comment:
This is repeated a lot; I'm thinking of passing a higher-order function instead; but that will have to wait.